packages/amesos2/src/Amesos2_TpetraMultiVecAdapter_def.hpp

```diff
@@ -211,12 +212,9 @@ namespace Amesos2 {
           const size_t lclNumRows = redist_mv.getLocalLength();
           for (size_t j = 0; j < redist_mv.getNumVectors(); ++j) {
             auto av_j = av(lda*j, lclNumRows);
-            //auto X_j = redist_mv.getVector(j);
             auto X_lcl_j_2d = redist_mv.template getLocalView<host_execution_space> ();
             auto X_lcl_j_1d = Kokkos::subview (X_lcl_j_2d, Kokkos::ALL (), j);
-            for ( size_t i = 0; i < lclNumRows; ++i ) {
-              av_j[i] = X_lcl_j_1d(i);
-            }
+            std::memcpy( &(av_j[0]), (X_lcl_j_1d.data()), sizeof( decltype(X_lcl_j_1d(0)) )*lclNumRows );
```

Comment:
The `sizeof` thing _will_ break for Stokhos and Sacado dynamic-length Scalar types.  How about this instead:
```c++
using val_type = Kokkos::ArithTraits<Scalar>::val_type; // whatever Scalar is
Kokkos::View<val_type*, Kokkos::HostSpace> avk (reinterpret_cast<val_type*> (av.getRawPtr ()), av.size ());
Kokkos::deep_copy (avk, _lcl_j_1d);
```

Reply:
@mhoemmen I didn't think about Fad types, thanks for suggested fix.